forte/data/data_store.py

```diff
@@ -234,6 +285,41 @@ def add_annotation_raw(
         # self.__entry_dict.
         raise NotImplementedError
 
+    def add_link_raw(
+        self, entry_type: str, parent_tid: int, child_tid: int
+    ) -> Tuple[int, int]:
+        r"""This function adds a link entry with `parent_tid` and `child_tid`
+        to the `entry_type` list, returns the `tid` and the 2-D index for the
+        inserted entry.
+
+        Args:
+            entry_type (str): Fully qualified name of this link.
+            parent_tid (int): `tid` of the parent entry.
+            child_tid (int): `tid` of the child entry.
+
+        Returns:
+            `tid` of the entry and its index in the entry list.
+
+        """
+        raise NotImplementedError
+
+    def add_group_raw(
+        self, entry_type: str, member_type: str
+    ) -> Tuple[int, int]:
+        r"""This function adds a group entry with `member_type` to the
+        `entry_type` list, returns the `tid` and the 2-D index for the
+        inserted entry.
+
+        Args:
+            entry_type (str): Fully qualified name of this group.
+            member_type (str): Fully qualified name of its members.
+
+        Returns:
+            `tid` of the entry and its index in the entry list.
```

Comment:
Explain the index in the entry list better.

Reply:
You should also create a function that takes the "index in the entry list" and return the actual data.

Reply:
yes